src/rime/dict/user_db.h

```diff
@@ -106,6 +106,12 @@ struct UserDbFormat {
   static const string snapshot_extension;
 };
 
+template <class BaseDb>
```

Comment:
This fails compilation with gcc.
I tried another way to get rid of the warnings but MSVC complained. It seems there is no way to make every compiler happy.
Just forget about the warnings for now.